cpp/src/arrow/array/array_test.cc

```diff
@@ -78,6 +78,26 @@ class TestArray : public ::testing::Test {
   MemoryPool* pool_;
 };
 
+TEST_F(TestArray, ValidateFullNullableList) {
+  auto f1 = field("f1", int32(), /*nullable=*/false);
+  auto ty = list(f1);
+  auto array = ArrayFromJSON(ty, "[[0, 1, 2, 3], null, [4, 5]]");
+  auto array_nested_null = ArrayFromJSON(ty, "[[0, 1, 2, null], [3,6], [4, 5]]");
+  // ASSERT_RAISES(Invalid, array->ValidateFull());
```

Comment:
Similar to the example below, we can use CopyBufferFromVector to manually assemble the array we want